access.py

```diff
@@ -0,0 +1,16 @@
+from flask_login import current_user
+from functools import wraps
+from flask import abort
+from user import User, Role
+
+def accesscheck(func):
+    @wraps(func)
+    def rolecheck(*args, **kwargs):  
+        
+        user = User(current_user.get_id())
+        if user.get_role() == Role.Writer:
```

Comment:
`current_user` is already an instance of your User class (because of your load_user method) so here we could just directly access the current_user value